src/deviation-check/deviation-check.ts

```diff
@@ -25,46 +27,67 @@ export const calculateMedian = (arr: bigint[]) => {
   return arr.length % 2 === 0 ? (nums[mid - 1]! + nums[mid]!) / 2n : nums[mid]!;
 };
 
-export const isDeviationThresholdExceeded = (onChainValue: bigint, deviationThreshold: bigint, apiValue: bigint) => {
-  const updateInPercentage = calculateDeviationPercentage(onChainValue, apiValue);
+export const isDeviationThresholdExceeded = (
+  onChainValue: bigint,
+  deviationThreshold: bigint,
+  apiValue: bigint,
+  deviationReference: bigint
+) => {
+  const updateInPercentage = calculateDeviationPercentage(onChainValue, apiValue, deviationReference);
 
-  return updateInPercentage > deviationThreshold;
+  return updateInPercentage >= deviationThreshold;
 };
 
 /**
- * Returns true when the on chain data timestamp is newer than the heartbeat interval.
+ * Returns true when the on-chain data is fresh enough not to be updated by the heartbeat.
  */
 export const isOnChainDataFresh = (timestamp: bigint, heartbeatInterval: bigint) =>
   BigInt(timestamp) > BigInt(Math.floor(Date.now() / 1000)) - heartbeatInterval;
 
+// Mirroring the logic of https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/extensions/BeaconSetUpdatesWithPsp.sol#L111
 export const isDataFeedUpdatable = (
   onChainValue: bigint,
   onChainTimestamp: bigint,
   offChainValue: bigint,
   offChainTimestamp: bigint,
   heartbeatInterval: bigint,
-  deviationThreshold: bigint
+  deviationThreshold: bigint,
+  deviationReference: bigint
 ): boolean => {
   // Check that fulfillment data is newer than on chain data. Update transaction with stale data would revert on chain,
-  // draining the sponsor wallet. See:
-  // https://github.com/api3dao/airnode-protocol-v1/blob/dev/contracts/dapis/DataFeedServer.sol#L121
+  // draining the sponsor wallet.
   if (offChainTimestamp <= onChainTimestamp) {
+    // It's possible that no beacon is updatable, but we may still update the data feed due to heartbeat.
     if (offChainTimestamp < onChainTimestamp) {
       logger.warn(`Off-chain sample's timestamp is older than on-chain timestamp.`);
     }
     return false;
   }
 
-  // Check that on chain data is newer than heartbeat interval
+  // At this point it's clear that the off-chain data is newer than the on-chain data and the on-chain data is
+  // uninitialized.
+  if (onChainTimestamp === 0n) return true;
```

Comment:
To totally stick with [contract](https://github.com/api3dao/airnode-protocol-v1/blob/main/contracts/api3-server-v1/extensions/BeaconSetUpdatesWithPsp.sol#L126), maybe we can do:
```suggestion
  if (onChainTimestamp === 0n && offChainTimestamp !== 0n ) return true;
```

Reply:
On a few lines above we check `if (offChainTimestamp <= onChainTimestamp)` so this is inferred.

Reply:
But the check can't really happen, because we only update beacons that can be updated. So it doesn't make sense to check whether the off-chain value is smaller on-chain. The only reason that exists is because the test has inconsistent mock data.

I'll adhere to the contract style.